packages/trader/src/index.ts

```diff
@@ -1,33 +1,136 @@
 import { config } from "dotenv";
-config();
+
 import retry from "async-retry";
+const { getWeb3 } = require("@uma/common");
+const { getAbi } = require("@uma/core");
+const {
+  GasEstimator,
+  Networker,
+  Logger,
+  createReferencePriceFeedForFinancialContract,
+  createTokenPriceFeedForFinancialContract,
+  waitForLogger,
+  delay,
+  DSProxyManager
+} = require("@uma/financial-templates-lib");
+
+const { RangeTrader } = require("./RangeTrader");
+const { createExchangeAdapter } = require("./exchange-adapters/CreateExchangeAdapter");
 import { TraderConfig } from "./TraderConfig";
+config();
+
+export async function run(logger: any, web3: any): Promise<void> {
+  try {
+    const getTime = () => Math.round(new Date().getTime() / 1000);
+    const config = new TraderConfig(process.env);
+
+    // If pollingDelay === 0 then the bot is running in serverless mode and should send a `debug` level log.
+    // Else, if running in loop mode (pollingDelay != 0), then it should send a `info` level log.
+    logger[config.pollingDelay === 0 ? "debug" : "info"]({
+      at: "Trader#index",
+      message: "Trader started 🚜",
+      financialContractAddress: config.financialContractAddress,
+      pollingDelay: config.pollingDelay,
+      errorRetries: config.errorRetries,
+      errorRetriesTimeout: config.errorRetriesTimeout,
+      tokenPriceFeedConfig: config.tokenPriceFeedConfig,
+      referencePriceFeedConfig: config.referencePriceFeedConfig,
+      exchangeAdapterConfig: config.exchangeAdapterConfig
+    });
+
+    // Load unlocked web3 accounts, get the networkId and set up price feed.
+    const networker = new Networker(logger);
+    const accounts = await web3.eth.getAccounts();
+
+    const gasEstimator = new GasEstimator(logger);
+
+    const dsProxyManager = new DSProxyManager({
+      logger,
+      web3,
+      gasEstimator,
+      account: accounts[0],
+      dsProxyFactoryAddress: config.dsProxyFactoryAddress,
+      dsProxyFactoryAbi: getAbi("DSProxyFactory"),
+      dsProxyAbi: getAbi("DSProxy")
+    });
+
+    await dsProxyManager.initializeDSProxy();
+
+    const [tokenPriceFeed, referencePriceFeed, exchangeAdapter] = await Promise.all([
```

Comment:
This bot relies on two main feeds: a `tokenPriceFeed` which is the synth market price (Uniswap feed, for example) and `referencePriceFeed` which is the "true" token price (desired) from the external market.